systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1075,499 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+        PasswordSecretSource passwordSecretSource = new PasswordSecretSource();
+        passwordSecretSource.setSecretName(kafkaUserSourceName);
+        passwordSecretSource.setPassword("password");
+
+        PasswordSecretSource passwordSecretTarget = new PasswordSecretSource();
+        passwordSecretTarget.setSecretName(kafkaUserTargetName);
+        passwordSecretTarget.setPassword("password");
+
+        // Initialize CertSecretSource with certificate and secret names for source
+        CertSecretSource certSecretSource = new CertSecretSource();
+        certSecretSource.setCertificate("ca.crt");
+        certSecretSource.setSecretName(KafkaResources.clusterCaCertificateSecretName(kafkaClusterSourceName));
+
+        // Initialize CertSecretSource with certificate and secret names for target
+        CertSecretSource certSecretTarget = new CertSecretSource();
+        certSecretTarget.setCertificate("ca.crt");
+        certSecretTarget.setSecretName(KafkaResources.clusterCaCertificateSecretName(kafkaClusterTargetName));
+
+        // Deploy client
+        resourceManager.createResource(extensionContext, KafkaClientsTemplates.kafkaClients(testStorage.getNamespaceName(), true, testStorage.getKafkaClientsName(), userSource, userTarget).build());
+        String kafkaClientsPodName = kubeClient().listPodsByPrefixInName(testStorage.getKafkaClientsName()).get(0).getMetadata().getName();
+
+        KafkaMirrorMaker2ClusterSpec sourceClusterWithScramSha512Auth = new KafkaMirrorMaker2ClusterSpecBuilder()
+            .withAlias(kafkaClusterSourceName)
+            .withBootstrapServers(KafkaResources.tlsBootstrapAddress(kafkaClusterSourceName))
+            .withNewKafkaClientAuthenticationScramSha512()
+                .withUsername(kafkaUserSourceName)
+                .withPasswordSecret(passwordSecretSource)
+            .endKafkaClientAuthenticationScramSha512()
+            .withNewTls()
+                .withTrustedCertificates(certSecretSource)
+            .endTls()
+            .addToConfig("config.storage.replication.factor", -1)
+            .addToConfig("offset.storage.replication.factor", -1)
+            .addToConfig("status.storage.replication.factor", -1)
+            .addToConfig("checkpoints.topic.replication.factor", "1")
+            .addToConfig("config.storage.replication.factor", 1)
+            .addToConfig("emit.checkpoints.enabled", "true")
+            .addToConfig("emit.checkpoints.interval.seconds", "1")
+            .addToConfig("heartbeats.topic.replication.factor", "1")
+            .addToConfig("offset.storage.replication.factor", 1)
+            .addToConfig("offset-syncs.topic.replication.factor", "1")
+            .addToConfig("refresh.groups.interval.seconds", "1")
+            .addToConfig("replication.factor", "1")
+            .build();
+
+        KafkaMirrorMaker2ClusterSpec targetClusterWithScramSha512Auth = new KafkaMirrorMaker2ClusterSpecBuilder()
```

Comment:
I think you can build new spec from existing one and just change values which you want

Reply:
I think that after my updates (now pushed yet) it does not make sense anymore, as there are only differences between Target/Source usages.